File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int low=0;
        int high=1;
        int maxSum=0;
        while(high<prices.size()){
            if(prices[low]>prices[high]){
                low=high;
            }
            else{
                int k=prices[high]-prices[low];
                maxSum=max(k,maxSum);
            }
            high++;
        }
        return maxSum;
    }
};

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        maxi=0
        low=0
        high=1
        while(high<len(prices)):
            if(prices[high]<prices[low]):
                low=high
            else:
                maxi=max(maxi,prices[high]-prices[low])

            high+=1
        return maxi